To-do list/Controllers/TaskListViewController.swift

```diff
@@ -63,21 +97,37 @@ extension TaskListViewController: UITableViewDelegate {
         let taskDetailsVC = TaskDetailsViewController.loadFromStoryboard(
             type: TaskDetailsViewController.self)
         taskDetailsVC.task = tasks[indexPath.row]
-        taskDetailsVC.delegate = self
         self.navigationController?.pushViewController(taskDetailsVC, animated: true)
     }
     
     func tableView(_ tableView: UITableView,
                    commit editingStyle: UITableViewCell.EditingStyle,
                    forRowAt indexPath: IndexPath) {
         if editingStyle == .delete {
-            tasks.remove(at: indexPath.row)
-            tableView.deleteRows(at: [indexPath], with: .fade)
             
+            taskApiHelper.deleteTask(id: "\(tasks[indexPath.row].id)") {[weak self] result in
+                guard let self = self else {
+                    return
+                }
+                var message = String()
+                if let result = result {
```

Comment:
again, duplicated result handling logic here. Make this logic generic, move it to separate func, and call it where needed (with injected dynamic params)